Static

yonicd opened the pull request:
# Pull Request

initial version of clindata package.

todo:
  - [x] construct api for caching
  - [x] write function documentation
  - [x] pass initial r cmd check locally
  - [x] write unit tests
  - [ ] write vignettes
  - [ ] pass complete r cmd check

Commit:
initial commit of bcva data constructed using targets

Commit:
add mermaid network plot genreated from the tar_mermaid for the bcva readme page

Commit:
add data description for bcva

Commit:
Add fev1 dataset

Commit:
generate api for fetching and listing datasets in package

Commit:
add more function to api

Commit:
refactor api functions

yonicd committed:
refactor package

yonicd committed:
remove mmrm-review from git tracking

yonicd committed:
remove mmrm_review from git tracking

yonicd committed:
update gitignore

yonicd committed:
fix linting for bcva-helpers.R

yonicd committed:
clean up lint in R functions
add first tests

yonicd committed:
add simulation code for fev data
refactor shared script for simulations

yonicd committed:
update bcva pipeline readme

yonicd committed:
add benchmark data for unit tests

yonicd committed:
replace rda with rds
add internal function to fetch benchmark data

yonicd commented:
@danielinteractive @wlandau I am still adding a few more [unit tests](https://github.com/openpharma/clindata/tree/static/tests/testthat) for the api, but the [simulation scripts](https://github.com/openpharma/clindata/tree/static/data-raw/pipelines) are ready to be looked at if you have time. 

I set up the target pipelines in a way that scenarios to generate various flavors of the data can be easily added by either us or other users.

The data from {mmrm} is currently being stored in [inst/benchmark](https://github.com/openpharma/clindata/tree/static/inst/benchmark)

yonicd committed:
add remaining unit tests for api

yonicd committed:
fix typo in bcva pipeline

yonicd committed:
cleanup documentation
refactor pkgdown.yml

danielinteractive commented on DESCRIPTION:
```diff
@@ -1,31 +1,23 @@
+Version: 0.0.1
+Date: 2023-11-09
+Authors@R: person("Jonathan","Sidi" , , "yoni@pinpointstrategies.io", role = c("aut", "cre"))
+Description: Open clinical data for openpharma packages.
```
```suggestion
Description: Open clinical data for use in R packages.
```
make it more general?

yonicd replied:
How about lean more into the idea that the package is a vehicle to transport reproducible simulations of clinical trial data? That has more potential to gain traction i think

wlandau replied:
What about non-simulated datasets from real trials? From our original discussion that led to this package, I think that would also be valuable for scope because it removes the burden from packages like `mmrm` and `brms.mmrm` from having to prove that the data are publicly available. Also, it is helpful to test on datasets which were not explicitly generated from any model we have. We would get a better sense of robustness and flexibility, especially under slight misspecification.

yonicd replied:
From the work on the package I have come to understand those weren't "real" data sets. They were realizations of simulated script. But I am to keep the option for real data if we come across it. Those datasets are being stored in the package (inst/benchmark) to allow for mmrm/brms.mmrm to depend on it. 

But I think in our last conversation in the group we talked about brms.mmrm can just use a seed on the simulation script from clindata simulated set and detach from the mmrm static dataset.

yonicd replied:
The main issue of storing real data iirc was the scalability of the size of the package. This led to the caching api.

danielinteractive commented on DESCRIPTION:
```diff
@@ -1,31 +1,23 @@
+Date: 2023-11-09
+Authors@R: person("Jonathan","Sidi" , , "yoni@pinpointstrategies.io", role = c("aut", "cre"))
+Description: Open clinical data for openpharma packages.
 License: Apache License 2.0 | file LICENSE
```
can already remove the LICENSE file, CRAN does not like if there are no modifications

yonicd replied:
would MIT file make more sense here?

danielinteractive replied:
Basically CRAN prefers to mention standard license here and not have any file. MIT or Apache both fine I think

yonicd replied:
MIT added

danielinteractive commented on R/api_cache.R:
```diff
@@ -0,0 +1,98 @@
+#' @title Data Caching API
+#' @description Functions to manage cache data across sessions.
+#' @param path character, path to cache directory, Default: cache_dir()
+#' @param \dots arguments passed to [list.files][base::list.files]
```
```suggestion
#' @param \dots arguments passed to [list.files()]
```
(base does not need to be prefixed)

danielinteractive commented on R/api_cache.R:
```diff
@@ -0,0 +1,98 @@
+#' @description Functions to manage cache data across sessions.
+#' @param path character, path to cache directory, Default: cache_dir()
+#' @param \dots arguments passed to [list.files][base::list.files]
+#' @param verbose logical, show message on success of [cache_destroy][cache_destroy]
```
```suggestion
#' @param verbose logical, show message on success of [cache_destroy()]
```
similar for the other places

danielinteractive commented on R/api_cache.R:
```diff
@@ -0,0 +1,98 @@
+#' @title Data Caching API
+#' @description Functions to manage cache data across sessions.
+#' @param path character, path to cache directory, Default: cache_dir()
```
```suggestion
#' @param path character, path to cache directory, Default: `cache_dir()`
```
would recommend to be consistent with this, so that you don't have to amend the wordlist for spelling with the code stuff

danielinteractive commented on R/api_cache.R:
```diff
@@ -0,0 +1,98 @@
+#'   Default: [cache_ls][cache_ls](full.names = TRUE)
+#' @param recursive boolean. Should directories be deleted recursively?
+#' @param force boolean. Should permissions be changed (if possible) to
+#'   allow the file or directory to be removed?
```
always include `@return` (even if nothing returned, then mention that) otherwise CRAN will tell you later

danielinteractive commented on R/api_cache.R:
```diff
@@ -0,0 +1,98 @@
+#'
+#'  # cleanup
+#'   cache_destroy()
+#' @rdname cache_rm
```
not needed?

danielinteractive commented on R/api_data.R:
```diff
@@ -0,0 +1,119 @@
+#' @param data character, name of data object
+#' @param filetype character, extension type of the data. Default: "rds"
+#' @param use_cache boolean, use the data stored in cache? Default: FALSE
+#' @return data.frame
```
I would recommend to explain what the meaning of the return value is, in addition to its type (generally and also here as one example)

danielinteractive commented on R/api_data.R:
```diff
@@ -0,0 +1,119 @@
+#'  read_data("bcva_data")
+#' @rdname data_api
+#' @export
+#' @importFrom tools file_path_sans_ext
```
would recommend to omit `importFrom` statements, as you don't need it - you prefix the usage below, so you access the function in its original `tools` namespace. just put `tools` in Imports in description file and you are good.

danielinteractive commented on R/api_data.R:
```diff
@@ -0,0 +1,119 @@
+        if(filetype == "rds"){
+          data_out <- readRDS(data_path)
+        }
+      }else{
```
good to run styler across the package 
```suggestion
      } else {
```

danielinteractive commented on R/data-fev.R:
```diff
@@ -0,0 +1,21 @@
+#'   - `FEV1_BL`: FEV1 at baseline (%).
+#'   - `FEV1`: FEV1 at study visits.
+#'   - `WEIGHT`: weighting variable.
+#'
```
include the `VISIT`* variables?

danielinteractive commented on README.md:
```diff
@@ -1,188 +1,48 @@
-# r.pkg.template
```
recommend to have a source `README.Rmd` and then knit that that `README.md`

yonicd replied:
Part of the documentation that I need to add for usage. I will convert to Rmd once there is something to render :)

danielinteractive commented on _pkgdown.yml:
```diff
@@ -15,6 +15,31 @@ template:
+      icon: fa-github
+      href: https://github.com/openpharma/clindata
+
+reference:
```
looks like a nice API!

danielinteractive commented on _targets.R:
```diff
@@ -0,0 +1,17 @@
+# Load packages required to define the pipeline:
```
maybe move this file into a `data-raw` folder or so? or is it customary to have it like this in the package root folder?

yonicd replied:
moved

danielinteractive commented on tests/testthat/test-api.R:
```diff
@@ -1,30 +1,71 @@
-  expected <- "Hello, Tim"
-  expect_identical(result, expected)
-})
+testthat::describe("caching logic", {
```
would recommend to load the `testthat` package and then avoid the many prefixes with its name in the tests

yonicd replied:
moved

yonicd replied:
old habits... adding ns to every fn as a reflex

danielinteractive commented on tests/testthat/test-data.R:
```diff
@@ -0,0 +1,35 @@
+testthat::describe("data api logic", {
```
interesting why not use `test_that()`?

yonicd replied:
it is just a convention i am used to. i like it more for creating hierarchies in reporting. It lets me get a better reporting structure when tests fail

https://testthat.r-lib.org/reference/describe.html

danielinteractive replied:
thanks, good to know! looks like a nice idea

danielinteractive reviewed (commented):
thanks @yonicd ! cool stuff

wlandau commented:
Interesting approach to generating fixed datasets with `targets` pipelines. I wonder though, since the `bcva` and `fev` datasets are simulated and the simulation code runs quickly, would it be easier to instead expose these as functions that the user could run to generate a dataset on the fly?

It's neat that you cover MCAR and MAR in `data-raw/pipelines/helpers.R`

yonicd commented:
@wlandau thanks. My initial intention was to make an example pipeline for people to "fork" and then add/extend their own simulation sets/designs. I thought that embedding it inside the package would create another hurdle for non pkg dev to join. Maybe the data agnostic fns can be put in the pkg?

The adv that i can see working through targets is that it enforces structure and is scalable and efficient (but you know that ;))

wlandau commented:
Yeah, I think if there are functions that generate datasets and do not require any fixed clinical data to start with, then I think those would make sense to document and export as usual.

yonicd committed:
resolve PR comments
move _targets into data-raw
add _targets.yaml to accomodate non root deployment

yonicd committed:
move data-raw/helpers.R into package
clean up license
update targets pipelines to depend on package

yonicd committed:
move ar1 matrix function to package
